Source/Extensions/Blazorise.DataGrid/ExtensionMethods.cs

```diff
@@ -45,14 +46,24 @@ public static string Localize( this ITextLocalizer textLocalizer, TextLocalizerH
             return textLocalizer[name, arguments];
         }
 
+
         /// <summary>
-        /// Checks if a type is a collection or a list.
+        /// Checks if a type is a collection.
         /// </summary>
         /// <param name="type">Type to check.</param>
-        /// <returns>True if <paramref name="type"/> is the collection or a list.</returns>
+        /// <returns>True if <paramref name="type"/> is a collection.</returns>
         public static bool IsCollection( this Type type )
-            => typeof( System.Collections.IList ).IsAssignableFrom( type )
-            || typeof( System.Collections.ICollection ).IsAssignableFrom( type )
-            || typeof( IEnumerable<> ).IsAssignableFrom( type );
+            => typeof( ICollection ).IsAssignableFrom( type )
+                || type.IsGenericCollection()
+                || type.IsGenericIEnumerable()
```

Comment:
Since every `ICollection<T>` is always an `IEnumerable<T>` (`interface ICollection<T> : IEnumerable<T>, IEnumerable` you don't need both the lines `type.IsGenericCollection()` and `type.IsGenericIEnumerable()`. One is wrong/superfluous.

Besides: you have different namings for both of the functions (one with `I`, one without)

Reply:
I guess it is superfluous haha, I was adding it as the tests were failing, thanks for noticing.  I'll take a look into changing it.

Personally think with the "I" is better named. 

I'll make these changes, thanks again.

Reply:
@WolfgangKluge so actually it wasn't quite superfluous as it was checking the type itself. It gets superfluous if you check for the interfaces, `Array.Exists( type.GetInterfaces(), IsGenericIEnumerable)` instead of `Array.Exists( type.GetInterfaces(), IsGenericICollection)`  but then we have the problem you had mentioned with `string` being an `IEnumerable<char>`. 
I see that as an off case, and I just added a check for `!= string`. 

God could it not be easier to check for a collection like type? haha

Reply:
Oh yes, you are right. Sorry